gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -714,6 +714,13 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
           s"${e.getMessage}, original sparkplan is " +
             s"${plan.getClass}(${plan.children.toList.map(_.getClass)})")
     }
+
+    if (TransformHints.isAlreadyTagged(plan) && TransformHints.isNotTransformable(plan)) {
```

Comment:
Maybe, we can move this part into a dedicated rule after  `AddTransformHintRule` (see [link](https://github.com/oap-project/gluten/blob/9d07d48e2f019bd1d3b1e36798c32f7cce62915c/gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala#L817)).

Reply:
Thanks, moved to a separate rule.